contrib/spring/README.md

```diff
@@ -0,0 +1,149 @@
+
+If the method throws an exception, the `Span` will be marked with a status of `Status.UNKNOWN`
+and the stack trace will be added to the span as an annotation.
+
+To enable the `@Traced` annotation, include the `CensusSpringAspect` bean.
+
+```xml
+  <!-- traces explicit calls to Traced -->
+  <bean id="censusAspect" class="io.opencensus.contrib.spring.aop.CensusSpringAspect">
+    <constructor-arg ref="tracer"/>
+  </bean>
+```
+
+#### Database Support
+
+The `opencensus-contrib-spring` package also includes support for tracing database
+calls.  When database support is included, all calls to `java.sql.PreparedStatement.execute*`
+will be wrapped in a Span in the same way that `@Traced` wraps methods.
+
+To enable database support, include the `CensusSpringSqlAspect` bean.
+
+```xml
+  <!-- traces all SQL calls -->
+  <bean id="censusSQLAspect" class="io.opencensus.contrib.spring.aop.CensusSpringSqlAspect">
+    <constructor-arg ref="tracer"/>
+  </bean>
+```
+
+#### Complete Spring XML configuration
+
+The following contains a complete spring xml file to configure `opencensus-contrib-spring` 
+with support for both `@Traced` and database connection tracing.
+
+**Note:** This example does not include the configuration of any exporters. That will 
+need to be done separately.
+
+**TBD:*** Include examples of spring with exporters.
+
+```xml
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xmlns:aop="http://www.springframework.org/schema/aop"
+       xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-3.2.xsd http://www.springframework.org/schema/aop http://www.springframework.org/schema/aop/spring-aop.xsd">
+
+  <aop:aspectj-autoproxy/>
+
+  <!-- traces explicit calls to Traced -->
+  <bean id="censusAspect" class="io.opencensus.contrib.spring.aop.CensusSpringAspect">
+    <constructor-arg ref="tracer"/>
+  </bean>
+
+  <!-- traces all SQL calls -->
+  <bean id="censusSQLAspect" class="io.opencensus.contrib.spring.aop.CensusSpringSqlAspect">
+    <constructor-arg ref="tracer"/>
+  </bean>
+
+  <!-- global tracer -->
+  <bean id="tracer" class="io.opencensus.trace.Tracing" factory-method="getTracer"/>
+</beans>
+```
+
+### Traced Usage 
+
+Once configured, you can use the `@Traced` annotation to indicate that a method should 
+be wrapped with a `Span`.  By default, `@Traced` will use the name of the method as the
+span name.  However, `@Traced` supports an optional name attribute to allow a custom
+span name to be specified.
+
+```java
+  @Traced()
+  void example1() {
+    // do work
+  }
+  
+  // a custom span name can also be provided to Traced
+  @Traced(name = "custom-span-name")
+  void example2() {
+    // do moar work
+  }
+```
+
+#### Notes
```

Comment:
Consider adding a section about JDK requirement (similar to https://github.com/census-instrumentation/opencensus-java/tree/master/exporters/trace/stackdriver#java-versions).